source/lac/trilinos_sparsity_pattern.cc

```diff
@@ -769,6 +790,14 @@ namespace TrilinosWrappers
             "Epetra indexes entries with a simple 'signed int'."));
       }
     AssertThrow(ierr == 0, ExcTrilinosError(ierr));
+
+    // Check consistency between the sizes set at the beginning and what
+    // Trilinos stores:
+    using namespace deal_II_exceptions::internals;
+    Assert(compare_for_equality(n_rows(), n_global_rows(*graph)),
+           ExcInternalError());
+    Assert(compare_for_equality(n_cols(), n_global_cols(*graph)),
+           ExcInternalError());
```

Comment:
AFAICT we do not support changing the sizes of the underlying Trilinos object outside of `reinit()`, which is why `n_rows()` does not need to be virtual.